dist/src/main/java/io/camunda/zeebe/shared/management/ExportingService.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.shared.management;
+
+import io.camunda.zeebe.gateway.admin.BrokerAdminRequest;
+import io.camunda.zeebe.gateway.impl.broker.BrokerClient;
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerClusterState;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ExportingService {
```

Comment:
:wrench: IMO, it would be good if we can move all "logic" into the gateway and only implement that user endpoint in dist module. This will make dist module as lean as possible, and all heavy logic + testing this in gateway.

Reply:
Sorry to chime in, but I had a similar idea about the `BackupRequestHandler`/`BackupApi`. I'm not sure it should be in the gateway, now that the actuators are neither broker nor gateway specific (since even a broker without a gateway can now make use of it). But I also don't know where they should go, we don't really have a module for this :sweat_smile: The only real dependency there is the broker client. We have thought about moving the broker client out of the gateway before, so maybe this would be something we could do (and it might help us when it comes to splitting responsibilities between teams, as the broker client is more a ZDT thing and the gateway logic a ZPT thing).

So all this to say, if you have some good idea where these things should go, go for it, otherwise I agree and we should centralize these things so they're easy to move later :+1:

Reply:
> otherwise I agree and we should centralize these things so they're easy to move later

Do you mean, it should stay in dist.shared?

Reply:
Well, the `BackupRequesthandler` is in the gateway, so I guess right now centralizing means the gateway? Though now that we have one there and one here, it's an even split :smile: I'm not for or against them, as I think neither the dist nor the gateway should contain these, so either works for me until we have another module for the broker client/broker API communication.